File: dotnet/Controllers/MainController.cs
using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;

namespace dotnet.Controllers
{
    [ApiController]
    public class MainController : ControllerBase
    {
        static readonly HttpClient client = new HttpClient();

        public MainController()
        {
        }

        [HttpGet("/hello")]
        public string Hello()
        {
            return "Hello World!";
        }

        [HttpGet("/fibo1")]
        public string Fibo1(string number)
        {
            return CalFibo1(int.Parse(number)).ToString();
        }

        [HttpGet("/fibo2")]
        public string Fibo2(string number)
        {
            int result = 0;
            int intNumber = int.Parse(number);

            if(intNumber <= 0) {
                result = 0;
            }else if(intNumber < 2) {
                result = 1;
            }else{
                int count = intNumber;
                int i, num1 = 0, num2 = 1;
                for (i = 2; i <= count; i++) {
                    result = num1 + num2;
                    num1 = num2;
                    num2 = result;
                }
            }

            return result.ToString();
        }

        [HttpGet("/io")]
        public async Task<string> IO()
        {
            HttpResponseMessage response = await client.GetAsync("http://nginx/index.html");
            string responseBody = await response.Content.ReadAsStringAsync();

            return responseBody.Length.ToString();
        }


        private int CalFibo1(int number){
            if(number <= 0) return 0;
            if(number < 2) return 1;

            return CalFibo1(number - 1) + CalFibo1(number - 2);
        }
    }
}

File: octane/app/Http/Controllers/MainController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;
use Illuminate\Routing\Controller;

class MainController extends Controller
{
    public function hello(){
        return "Hello World!";
    }

    public function fibo1(Request $request){
        return $this->calFibo1($request->get('number'));
    }

    public function fibo2(Request $request){
        $result = 0;
        $number = $request->get('number');
        if($number <= 0){
            $result = 0;
        }elseif($number <= 2){
            $result = 1;
        }else{
            $count = $number;
            $num1 = 0;
            $num2 = 1;

            for ($i = 2; $i <= $count; $i++){
                $result = $num1 + $num2;
                $num1 = $num2;
                $num2 = $result;
            }
        }

        return $result;
    }

    public function io(){
        return strlen(file_get_contents("http://nginx/index.html"));
    }

    private function calFibo1($number){
        if($number <= 0) 
            return 0;
        if($number < 2) 
            return 1;

        return $this->calFibo1($number - 1) + $this->calFibo1($number - 2);
    }
}
